pkg/apis/serving/metadata_validation.go

```diff
@@ -30,7 +31,18 @@ import (
 // resources is correct.
 func ValidateObjectMetadata(meta metav1.Object) *apis.FieldError {
 	return apis.ValidateObjectMetadata(meta).
-		Also(autoscaling.ValidateAnnotations(meta.GetAnnotations()).ViaField("annotations"))
+		Also(autoscaling.ValidateAnnotations(meta.GetAnnotations()).
+			Also(validateKnativeAnnotations(meta.GetAnnotations())).
+			ViaField("annotations"))
+}
+
+func validateKnativeAnnotations(annotations map[string]string) (errs *apis.FieldError) {
+	for key := range annotations {
+		if strings.HasPrefix(key, GroupNamePrefix) && key != UpdaterAnnotation && key != CreatorAnnotation && key != RevisionLastPinnedAnnotationKey {
```

Comment:
`serving.knative.dev/visibility` unfortunately has the same prefix. It should be in this list (at least for Services and Routes)

https://knative.dev/docs/serving/cluster-local-route/